src/signature.rs

```diff
@@ -66,6 +67,15 @@ pub use rsa::{
     RSA_PKCS1_3072_8192_SHA384_VERIFY,
 };
 
+/// A signature holder struct.
+//
+// The `Signature` struct is a just a wrapper around a `u8` array. It is
+// only used for signing operations, not for verification operations.
+pub struct Signature {
+    /// The array holding the signature data.
```

Comment:
Please add a comment explaining why `value` is public and also mark this:
```
#[doc(hidden)]
```

Also, please add this function:
```
/// Returns the signature's encoded value.
pub fn as_slice(&self) -> &'a [u8] { &self.value[..] }
```